bin/deploy-vhost

```diff
@@ -478,6 +449,46 @@ sub update_vcspath_symlink {
     chdir_verbose(__LINE__, $old_cwd);
 }
 
+# Generate daemon init scripts from templates
+sub create_daemons {
+  foreach my $daemon (keys %{$conf->{'daemons'}}) {
+    my $daemon_mugly_file = $conf->{'daemons'}->{$daemon};
+    # See if there's an ugly file in the servers dir
+    $daemon_mugly_file  =~ m#^(.*?).ugly$#;
+    my $name_part = $1;
+    my $daemon_mugly = mktemp("/tmp/daemon-mugly-file-XXXXXXXX");
+    my $servers_dir_ugly_file = get_conf_ugly_file_from_servers_dir($name_part);
+
+    if (-e $servers_dir_ugly_file) {
+      shell("cat $servers_dir_ugly_file >$daemon_mugly");
+    } else {
+      chdir_verbose(__LINE__, "$vhost_dir/$vcspath_install");
+      my $git_ref;
+      my $conf_dir;
+      if ( $conf->{git_ref} ) {
+          $git_ref = $conf->{git_ref};
+          $conf_dir = $conf->{conf_dir}->[0]
+      } elsif ( $conf->{private_git_ref} ) {
+          $git_ref = $conf->{private_git_ref};
+          $conf_dir = $conf->{private_conf_dir}->[0]
+      }
+
+      shell("git show $git_ref:" . $conf_dir . "/$daemon_mugly_file >$daemon_mugly");
+    }
+
+    my ($fh, $filename) = tempfile();
```

Comment:
Again, I realise this is what it was before, but I don't understand why we get a file handle, then in the next line create a new file handle…? Weird.